otwcode/otwarchive
# AO3-5593 Rewrite definition of approved_collections.

## Issue

https://otwarchive.atlassian.net/browse/AO3-5593

## Purpose

This PR modifies the definition of `approved_collections`, `user_approved_collections`, and `rejected_collections` so that they no longer use the `Collection.approved`, `Collection.user_approved`, and `Collection.rejected` scopes, each of which has a call to `includes(:collection_items)` that results in excess database queries.

These scopes were introduced in #2927 for the purpose of simplifying definitions, but since this PR removes any dependence on those scopes, I have removed them.

I also modified the three `includes()` statements in `WorksController` to preload `approved_collections` instead of `collections: [:collection_items]`.

## Testing Instructions

1. Go into the rails console.
2. Make sure that `ActiveRecord::Base.logger.level` is set low enough that you see SQL queries.
3. Find a work with at least one approved collection.
4. Call `Work.find(id).approved_collections`, and make sure that it results in exactly two database queries.
